pkg/controller/container-runtime-config/helpers.go

```diff
@@ -0,0 +1,219 @@
+	for _, c := range mc.Spec.Config.Storage.Files {
+		if c.Path == "/etc/crio/crio.conf" {
+			return &c, nil
+		}
+	}
+	return nil, fmt.Errorf("Could not find CRI-O Config")
+}
+
+func getManagedKey(pool *mcfgv1.MachineConfigPool, config *mcfgv1.ContainerRuntimeConfig) string {
+	return fmt.Sprintf("99-%s-%s-containerruntime", pool.Name, pool.ObjectMeta.UID)
+}
+
+func wrapErrorWithCondition(err error, args ...interface{}) mcfgv1.ContainerRuntimeConfigCondition {
+	var condition *mcfgv1.ContainerRuntimeConfigCondition
+	if err != nil {
+		condition = mcfgv1.NewContainerRuntimeConfigCondition(
+			mcfgv1.ContainerRuntimeConfigFailure,
+			v1.ConditionFalse,
+			fmt.Sprintf("Error: %v", err),
+		)
+	} else {
+		condition = mcfgv1.NewContainerRuntimeConfigCondition(
+			mcfgv1.ContainerRuntimeConfigSuccess,
+			v1.ConditionTrue,
+			"Success",
+		)
+	}
+	if len(args) > 0 {
+		format, ok := args[0].(string)
+		if ok {
+			condition.Message = fmt.Sprintf(format, args[:1]...)
+		}
+	}
+	return *condition
+}
+
+func updateStorageConfig(data []byte, internal *mcfgv1.ContainerRuntimeConfiguration) ([]byte, error) {
+	tomlConf := new(tomlConfigStorage)
+	if _, err := toml.Decode(string(data), tomlConf); err != nil {
+		return nil, fmt.Errorf("error unmarshalling crio config: %v", err)
+	}
+
+	if internal.OverlaySize != "" {
+		tomlConf.Storage.Options.Size = internal.OverlaySize
+	}
+
+	var newData bytes.Buffer
+	encoder := toml.NewEncoder(&newData)
+	if err := encoder.Encode(*tomlConf); err != nil {
+		return nil, err
+	}
+	return newData.Bytes(), nil
+}
+
+func updateCRIOConfig(data []byte, internal *mcfgv1.ContainerRuntimeConfiguration) ([]byte, error) {
+	tomlConf := new(tomlConfigCRIO)
+	if _, err := toml.Decode(string(data), tomlConf); err != nil {
+		return nil, fmt.Errorf("error unmarshalling crio config: %v", err)
+	}
+
+	if internal.PidsLimit > 0 {
+		tomlConf.Crio.Runtime.PidsLimit = internal.PidsLimit
+	}
+	if internal.LogSizeMax != 0 {
+		tomlConf.Crio.Runtime.LogSizeMax = internal.LogSizeMax
+	}
+	if internal.LogLevel != "" {
+		tomlConf.Crio.Runtime.LogLevel = internal.LogLevel
+	}
+	if internal.InfraImage != "" {
+		tomlConf.Crio.Image.PauseImage = internal.InfraImage
+	}
+
+	var newData bytes.Buffer
+	encoder := toml.NewEncoder(&newData)
+	if err := encoder.Encode(*tomlConf); err != nil {
+		return nil, err
+	}
+
+	return newData.Bytes(), nil
+}
+
+func validateUserContainerRuntimeConfig(cfg *mcfgv1.ContainerRuntimeConfig) error {
+	if cfg.Spec.ContainerRuntimeConfig == nil {
+		return nil
+	}
+	ctrcfgValues := reflect.ValueOf(*cfg.Spec.ContainerRuntimeConfig)
+	if !ctrcfgValues.IsValid() {
+		return fmt.Errorf("ContainerRuntimeConfig is not valid")
+	}
+
+	ctrcfg := cfg.Spec.ContainerRuntimeConfig
+	if ctrcfg.PidsLimit > 0 && ctrcfg.PidsLimit < 20 {
+		return fmt.Errorf("Invalid PidsLimit %q, cannot be less than 20", ctrcfg.PidsLimit)
+	}
+
+	if ctrcfg.LogSizeMax > 0 && ctrcfg.LogSizeMax < 8193 {
```

Comment:
We should use a const for `8192` and do a <= here.

Reply:
Fixed